fix: 20585 - Web - Split Bill - Money request confirmation list stays highlighted after press

Skalakid opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

### Fixed Issues
<!---
1. Please replace GH_LINK with a URL link to the GitHub issue this Pull Request is fixing.
2. Please replace PROPOSAL: GH_LINK_ISSUE(COMMENT) with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>
$ https://github.com/Expensify/App/issues/<number-of-the-issue(comment)>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/App/issues/20585
PROPOSAL: https://github.com/Expensify/App/issues/20585#issuecomment-1593021403


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->
1. Log in
2. Create group chat (click floating `+`  button > `New group` > select some users)
3. Enter group chat
4. Click `+` button that is in the `TextInput`
5. Select `Split bill`
6. Enter any amount of money
7. Start selecting and unselecting people in the list, and check if only the first unselected user is highlighted

- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->
Same as Tests
### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
8. Upload an image via copy paste
9. Verify a modal appears displaying a preview of that image
--->
Same as Tests
- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/39538890/71c1c969-89d1-4c4b-9970-d05aeabdeaec


</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->
N/A
</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->
N/A
</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->
N/A
</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->
N/A
</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->
N/A
</details>

Skalakid committed:
fixed issue with items staying higlighted

Skalakid commented on src/components/Pressable/PressableWithFeedback.js:
```diff
@@ -66,17 +66,15 @@ const PressableWithFeedback = forwardRef((props, ref) => {
                     });
                 });
             }}
+            hoverStyle={props.hoverStyle}
+            pressStyle={props.pressStyle}
+            focusStyle={props.focusStyle}
```
When creating these Pressable components I think we wanted to separate styles from GenericPressable and use them only in child component - OpacityView

Skalakid replied:
By passing them, to the parent component, we can easily fix [the bug with blue borders](https://github.com/Expensify/App/pull/19612#issuecomment-1569876479)

s77rt replied:
Hmm I think the current implementation is the correct now, not sure why we did it that way but I think probably to support `Button`. This change breaks `Button` styles.

Didn't dive into the problem but seems like this should be fixed on OptionRow level.

@Skalakid I see you are not assigned on the linked issue yet. Were you assigned somewhere else?

Skalakid replied:
No, I wasn't assigned somewhere else, I've added the fix as it is a regression from my previous PR https://github.com/Expensify/App/pull/19612 and @Christinadobrzyn asked me for some help.

Btw, can you provide more information on why this change brakes `Button` styles, please?

s77rt replied:
The change itself is a breaking change, styles are no longer applied as they were. To reproduce the bug just hover any button e.g. Settings -> Payments -> "Add payment method" button

| Before | After |
|--------|-------|
|   <img src="https://github.com/Expensify/App/assets/16493223/33d78ccb-0a46-4555-86cb-1f534877ffd0" />     |    <img src="https://github.com/Expensify/App/assets/16493223/197a5b8d-7d16-4f5a-9f1d-5cca0dcc9380" />   |

Skalakid committed:
fixed bugs with styling after PressableWithFeedback structure change

Skalakid commented:
@s77rt I've added some changes, these should fix styling issues

Skalakid committed:
fixed passing onHoverIn, onHoverOut etc functions to Pressable

Skalakid committed:
removed ommiting pressable styles

Skalakid commented:
I made changes in `PressableWithFeedback` to don't break the previous [fix of the focus blue borders](https://github.com/Expensify/App/pull/19612#issuecomment-1569876479). The problem is that we can focus on clickable elements using the keyboard's `tab` button. When we select items using `tab`, elements are being focused but with an additional blue border. Previously we just changed focus styles to be the same as hover styles to just hide the blue border. But now we want to delete focusStyle because it causes the bug with highlighting from this issue.
When we just remove `focusStyle` it will cause something like this:

https://github.com/Expensify/App/assets/39538890/7c6bda88-db55-467f-a258-2899bae3cd44

The problem is that we are focussing on pressable but the `OpacityView` styles are above the blue border and a user can't see it. So I changed the order of the components, and now the Pressable that we are focussing on is the child component. So we can see the blue border and all other styles at the same time! 


https://github.com/Expensify/App/assets/39538890/f31df5dd-f3fa-47cf-ad5d-308c31ba5a34

s77rt commented:
Bug: Buttons not being dimmed on press.

Skalakid committed:
fixed bug with dimming

Julesssss reviewed (dismissed):
Tested again and I see the button dimming on web now 👍

s77rt commented on src/components/OpacityView.js:
```diff
@@ -49,11 +47,7 @@ function OpacityView(props) {
-            <View style={StyleUtils.parseStyleAsArray(props.style)}>{props.children}</View>
-        </Animated.View>
-    );
+    return <Animated.View style={[props.style, opacityStyle]}>{props.children}</Animated.View>;
```
This is against the styling guide. We should still `parseStyleAsArray` and deconstruct the styles.
```js
<Animated.View style={[opacityStyle, ...StyleUtils.parseStyleAsArray(props.style)]}>{props.children}</Animated.View>;
```

Skalakid committed:
added parseStyleAsArray to styles

s77rt reviewed (approved):
LGTM! :rocket: 

cc @Julesssss

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5329214929) to staging by https://github.com/Julesssss in version: 1.3.30-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

s77rt commented on src/components/Pressable/PressableWithFeedback.js:
@Skalakid Just found out we are doing something wrong here. Can you please raise a quick follow up to fix this.
1. The rest props we pass to `GenericPressable` should be the first not to overwrite our props (onHoverIn, onPress, etc)
2. The omittedProps should only be `wrapperStyle`

Something like that
```js
diff --git a/src/components/Pressable/PressableWithFeedback.js b/src/components/Pressable/PressableWithFeedback.js
index 2949683734..3defd7d640 100644
--- a/src/components/Pressable/PressableWithFeedback.js
+++ b/src/components/Pressable/PressableWithFeedback.js
@@ -7,7 +7,7 @@ import GenericPressablePropTypes from './GenericPressable/PropTypes';
 import OpacityView from '../OpacityView';
 import variables from '../../styles/variables';
 
-const omittedProps = ['wrapperStyle', 'onHoverIn', 'onHoverOut', 'onPressIn', 'onPressOut'];
+const omittedProps = ['wrapperStyle'];
 
 const PressableWithFeedbackPropTypes = {
     ...GenericPressablePropTypes.pressablePropTypes,
@@ -55,6 +55,8 @@ const PressableWithFeedback = forwardRef((props, ref) => {
         >
             <GenericPressable
                 ref={ref}
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...rest}
                 disabled={disabled}
                 onHoverIn={() => {
                     setIsHovered(true);
@@ -85,8 +87,6 @@ const PressableWithFeedback = forwardRef((props, ref) => {
                         });
                     });
                 }}
-                // eslint-disable-next-line react/jsx-props-no-spreading
-                {...rest}
             >
                 {(state) => (_.isFunction(props.children) ? props.children(state) : props.children)}
             </GenericPressable>
```

Skalakid replied:
@s77rt I will handle this and create PR with the follow-up

jasperhuangg replied:
@Skalakid Just wanted to let you know that this proposal created a regression: https://github.com/Expensify/App/issues/21354

s77rt replied:
Hmm wasn't this a bug even before this PR? This PR hide some bugs (and some features) we had fix that in https://github.com/Expensify/App/pull/21310

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5348093925) to production by https://github.com/luacmartins in version: 1.3.30-5 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅
